crates/runix/src/flake_ref/mod.rs

```diff
@@ -286,5 +490,126 @@ pub(super) mod tests {
+        File::create(basic.join("flake.nix")).unwrap();
+
+        let path = pathdiff::diff_paths(&basic, env::current_dir().unwrap()).unwrap();
+
+        assert_eq!(
+            FlakeRef::resolve_local(&path.to_string_lossy())
+                .unwrap()
+                .to_string(),
+            Url::parse(&format!("path:{}", basic.to_string_lossy()))
+                .unwrap()
+                .to_string()
+        )
+    }
+
+    #[test]
+    fn test_resolve_absolute_git_local() {
+        let flake_test_dir = tempfile::tempdir().unwrap();
+
+        let git_dir = flake_test_dir
+            .path()
+            .canonicalize()
+            .unwrap()
+            .join("withgit");
+        fs::create_dir_all(&git_dir).unwrap();
+        fs::create_dir_all(git_dir.join(".git")).unwrap();
+        File::create(git_dir.join("flake.nix")).unwrap();
+
+        let git_dir_string = git_dir.to_string_lossy();
+
+        assert_eq!(
+            FlakeRef::resolve_local(&git_dir_string)
+                .unwrap()
+                .to_string(),
+            Url::parse(&format!("git+file:{git_dir_string}"))
+                .unwrap()
+                .to_string()
+        )
+    }
+
+    #[test]
+    fn test_resolve_absolute_git_local_nested() {
+        let flake_test_dir = tempfile::tempdir().unwrap();
+
+        let flake_dir_name = "inner";
+
+        let git_dir = flake_test_dir
+            .path()
+            .canonicalize()
+            .unwrap()
+            .join("withgit");
+        fs::create_dir_all(&git_dir).unwrap();
+        fs::create_dir_all(git_dir.join(".git")).unwrap();
+
+        let flake_root = git_dir.join(flake_dir_name);
+        fs::create_dir_all(&flake_root).unwrap();
+
+        File::create(flake_root.join("flake.nix")).unwrap();
+
+        let git_dir_string = git_dir.to_string_lossy();
+
+        assert_eq!(
+            FlakeRef::resolve_local(&flake_root.to_string_lossy())
+                .unwrap()
+                .to_string(),
+            Url::parse(&format!("git+file:{git_dir_string}?dir={flake_dir_name}"))
+                .unwrap()
+                .to_string()
+        )
+    }
+
+    #[test]
+    fn test_resolve_relative_git_local() {
+        let flake_test_dir = tempfile::tempdir().unwrap();
+
+        let git_dir = flake_test_dir
+            .path()
+            .canonicalize()
+            .unwrap()
+            .join("withgit");
+        fs::create_dir_all(&git_dir).unwrap();
+        fs::create_dir_all(git_dir.join(".git")).unwrap();
+        File::create(git_dir.join("flake.nix")).unwrap();
+
+        let relative_path = pathdiff::diff_paths(&git_dir, env::current_dir().unwrap()).unwrap();
+
+        assert_eq!(
+            FlakeRef::resolve_local(relative_path.to_string_lossy())
+                .unwrap()
+                .to_string(),
+            Url::parse(&format!("git+file:{}", git_dir.to_string_lossy()))
+                .unwrap()
+                .to_string()
+        )
```

Comment:
There isn't coverage for any of the error edge cases - crossing git boundary, crossing device boundary, inconsistent dir param

Reply:
Don't know how to account for device boundaries but i added tests for git boundary and dir param